actioncable/app/javascript/action_cable/connection.js

```diff
@@ -128,6 +128,7 @@ Connection.prototype.events = {
   message(event) {
     if (!this.isProtocolSupported()) { return }
     const {identifier, message, reason, reconnect, type} = JSON.parse(event.data)
+    this.monitor.recordPing()
```

Comment:
I think the answer is "No concern" but I still want to ask it out loud:
Is there any concern with calling `recordPing()` twice? For example if the message type is a `message_types.ping` itself we will call it twice. I think there is no concern because it will just set the `pingedAt` value unless I missed something else.

Also similarly, any reason for us to consider removing `this.recordPing() ` from the `recordConnect()` itself? We only call `recordConnect()` from this statement so the second `recordPing()` call is redundant. I guess it's okay to keep it because it can't break anything and unlikely will be a noticeable performance improvement. Also keeping recordPing()` in `recordConnect()` will ensure that pings are recorded in cases when `recordConnect()` is called from other context
 
https://github.com/rails/rails/blob/abde74c118a2391c90f4c8ac197a9628ef2acf9b/actioncable/app/javascript/action_cable/connection_monitor.js#L46

Reply:
These are good questions.

I think, it would be better to rename `recordPing` to `recordMessage` and leave it only here (thus, remove from `recordConnect` and `case message_types.ping`). Double calls are not problematic but they make understanding the codebase more confusing.

Reply:
Fair! I feel a little nitpicky so I'm going to mention that I wouldn't mind seeing this PR merged as is to deliver the fix as soon as possible with a refactoring being done afterwards.